WEST MIDLANDS | Roja Alagurajan| MODULE-ONBOARDING | WEEK 1| ITP-MAY-2025 | WIREFRAME PROJECT |

alagurajan-roja opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist
This pull request improves the structure, alignment, and styling of the “Web Development Basics” page. It fixes layout issues, improves spacing consistency, corrects color codes, and ensures a responsive, accessible design.

Fixed CSS syntax issues (extra } and invalid color code).

Improved layout using CSS Grid for better alignment.

Added proper borders and spacing to articles.

Cleaned up HTML content for clarity and consistency.
I made a changes format our code for better readability and consistency

Why have three different classes if they share the same CSS properties?

They share many common styles, so those are written once on the article selector to avoid repetition. The different classes allow small, unique differences, like border color, to be applied individually. Using separate classes makes it easier to maintain and update styles specific to each article type without affecting others.


Code passes W3C validation with no errors or warnings.

Achieved 100% Accessibility score in Lighthouse.

## Questions

Ask any questions you have for your reviewer.

alicagatay commented on Wireframe/style.css:
```diff
@@ -69,21 +36,72 @@ main {
+}
+
+
+.read-more {
```
While styling the components of a web page, how do we decide the colours of different components in the page? What factors do we need to consider when colouring?

Or, for example, why did you choose the following colours for the buttons of your page?

<img width="253" alt="image" src="https://github.com/user-attachments/assets/a33e97c9-11b1-417a-b988-a761ffa7b149" />




Have a think about these questions. You can also have a look at the following resources to learn more about this topic:

1. https://www.flux-academy.com/blog/ultimate-guide-to-choosing-colors-for-web-design

2. https://accessbydesign.uk/why-colour-contrast-is-important-in-accessible-web-design/#:~:text=The%20benefits%20of%20maintaining%20colour,devices%20with%20lower%20quality%20screens.

3. https://www.the215guys.com/blog/color-theory-accessibility-web-design/

alagurajan-roja replied:
@alicagatay
Hi alicagatay,
I have corrected my code. Kindly check. And let me know. if i need to change anything

alagurajan-roja replied:
hi @cjyuan ,
I have checked and made the necessary fixes. The layout has been updated, and I’m happy to share that I’ve achieved a 100% score in Lighthouse, including accessibility. Please review and let me know if any further changes are needed.

alicagatay commented on Wireframe/index.html:
```diff
@@ -1,33 +1,56 @@
-      <p>
-        This is the default, provided code and no changes have been made yet.
-      </p>
+       <h1>Web Development Basics</h1>
+      <p>Web development is the process of creating websites and web applications for the internet and its involve front-end and back-end development.</p>
```
In the given wireframe image, the header and description are centred on the page, but in your example, they are positioned on the left side of the page. Can you think of a way to change the layout of these texts?

The given wireframe:
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/0ddf132c-c26b-4f74-8f6e-14197c392c96" />

Your example:
<img width="1481" alt="image" src="https://github.com/user-attachments/assets/2708ac9a-cad8-4f92-a8f7-1f29991c4318" />

alicagatay commented on Wireframe/index.html:
```diff
@@ -1,33 +1,56 @@
-        This is the default, provided code and no changes have been made yet.
-      </p>
+     <footer>
+      <p>Created by Roja Alagurajan. Last updated: May 15, 2025.</p>
```
While scrolling through the webpage, I have noticed that the footer stays at the bottom of the screen, which causes it to collide with other components on the webpage. Please take a look at how this behaviour can be addressed.

A couple examples of what I mean:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/95b21b21-e93f-4059-98be-a429494ad7e6" />

<img width="505" alt="image" src="https://github.com/user-attachments/assets/af8e103e-438e-455e-8f44-4dc6f4909b02" />

alagurajan-roja replied:
hi @cjyuan 
I have corrected my code . Kindly check . And let me know . if i need to change anything

alicagatay reviewed (commented):
In overall, you have made a great start, congratulations. There is only some fixes that you need to make in terms of layout, which I have addressed in my comments below, but other than that, this was a great start 😁.

alagurajan-roja commented:
@alicagatay
Hi alicagatay,
I have corrected my code . Kindly check . And let me know . if i need to change anything.


Regards ,
Roja alagurajan

cjyuan commented:
@alicagatay Since the deadline to submit Step 1 is June 6, I'm stepping in to help move this PR forward.

alagurajan-roja commented:
Hi @cjyuan  
I have corrected my code. Kindly check .

cjyuan commented on Wireframe/style.css:
```diff
@@ -69,21 +37,84 @@ main {
+}
+
+/* ====== Special Classes ====== */
+.Wireframe ,
+.git,
+.Readfile {
```
Why introduce three different classes but they all share the same CSS property? Why not just use one class and have all `<article>`  share the same class?

alagurajan-roja replied:
Hi @cjyuan ,
This pull request improves the structure, alignment, and styling of the “Web Development Basics” page. It fixes layout issues, improves spacing consistency, corrects color codes, and ensures a responsive, accessible design.

Fixed CSS syntax issues (extra } and invalid color code).

Improved layout using CSS Grid for better alignment.

Added proper borders and spacing to articles.

Enhanced "Read more" button styling and hover effects.

Cleaned up HTML content for clarity and consistency.



**Why have three different classes if they share the same CSS properties?**

They share many common styles, so those are written once on the article selector to avoid repetition. The different classes allow small, unique differences, like border color, to be applied individually. Using separate classes makes it easier to maintain and update styles specific to each article type without affecting others.


Please let me know any changes 


Regars,
Roja

cjyuan reviewed (commented):
Code is not error-free, and webpage matches the wireframe layout and appearance. Good job.

1. Can for improve the indentation of your HTML code? 

VSCode's "Format Document" feature can help us format our code for better readability and consistency, including breaking a long line of text into shorter lines of text.
To use the feature, right-click inside the code editor and select the option.
Please note that if there are syntax errors in the code, the "Prettier" extension may not format HTML code properly.

2. To follow best practices, in your PR description, can you provide a brief description (under the "Changelist" section) summarizing the purpose of the PR and the changes you’ve made?
![image](https://github.com/user-attachments/assets/f510bc87-2e14-4e47-853b-65a7453d7b41)

alagurajan-roja commented:
Hi @cjyuan ,
This pull request improves the structure, alignment, and styling of the “Web Development Basics” page. It fixes layout issues, improves spacing consistency, corrects color codes, and ensures a responsive, accessible design.

Fixed CSS syntax issues (extra } and invalid color code).

Improved layout using CSS Grid for better alignment.

Added proper borders and spacing to articles.

Cleaned up HTML content for clarity and consistency.
I made a changes format our code for better readability and consistency

Why have three different classes if they share the same CSS properties?

They share many common styles, so those are written once on the article selector to avoid repetition. The different classes allow small, unique differences, like border color, to be applied individually. Using separate classes makes it easier to maintain and update styles specific to each article type without affecting others.

Please let me know any changes

Regars,
Roja

cjyuan commented:
Can you also addressed the issues about code indentation and brief description of PR in my previous feedback?

cjyuan commented:
Changes look good.
